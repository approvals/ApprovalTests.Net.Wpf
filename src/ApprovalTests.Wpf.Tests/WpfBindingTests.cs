using System;
using System.Text.RegularExpressions;
using System.Windows.Controls;
using System.Windows.Data;
using ApprovalTests;
using ApprovalTests.Reporters;
using ApprovalTests.Wpf;
using ApprovalUtilities.Utilities;
using Xunit;

[UseReporter(typeof(DiffReporter))]
public class WpfBindingTests
{
    [Fact]
    public void TestFailedBindings()
    {
        Exception e = null;
        StaRunner.Start(() =>
        {
            var viewModel = new TestViewModel();
            var myBinding = new Binding(TestViewModel.MyPropertyPropertyName + "BOGUS")
            {
                Source = viewModel
            };
            e = ExceptionUtilities.GetException(
                () => WpfBindingsAssert.BindsWithoutError(viewModel,
                    () =>
                    {
                        var textBox = new TextBox();
                        textBox.SetBinding(TextBox.TextProperty, myBinding);
                        return textBox;
                    }));
        });
        Approvals.Verify(e.Message, s => Regex.Replace(s, @"\(HashCode=\d+\)", "(Hashcode)"));
    }
}