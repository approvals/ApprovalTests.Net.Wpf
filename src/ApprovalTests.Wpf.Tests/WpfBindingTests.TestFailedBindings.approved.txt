System.Windows.Data Error: 40 : BindingExpression path error: 'MyPropertyBOGUS' property not found on 'object' ''TestViewModel' (Hashcode)'.
	BindingExpression:Path=MyPropertyBOGUS
	 DataItem='TestViewModel' (Hashcode)
	 target element is 'TextBox' (Name='')
	 target property is 'Text' (type 'String')
